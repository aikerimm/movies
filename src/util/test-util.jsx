import { configureStore, createAsyncThunk } from '@reduxjs/toolkit';
import moviesReducer from './moviesSlice.jsx';
import { Provider } from 'react-redux';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';

export const testMovie = {
  id: 123,
  title: 'Zootopia',
  vote_average: 9.8,
  runtime: 120,
  genres: ['Drama', 'Comedy'],
  release_date: '1994-01-01',
  overview: 'Lorem Ipsum Some',
  poster_path: '',
};

export const wrapWithReduxAndRouter = (component, initialPath) => {
  const testStore = configureStore({
    reducer: { movies: moviesReducer },
    preloadedState: {
      movies: {
        data: { totalAmount: 1, data: [testMovie], offset: 0, limit: 12 },
        totalAmount: 0,
      },
    },
  });

  const testFetchMovie = createAsyncThunk('movies/fetchMovie', async () => {
    return Promise.resolve(testMovie);
  });

  const testFetchMovies = createAsyncThunk('movies/fetchMovies', async () => {
    return Promise.resolve([testMovie]);
  });

  const testLoader = () => {
    return {
      moviesThunk: testFetchMovies,
      movieThunk: testFetchMovie,
    };
  };

  const testRouter = createMemoryRouter(
    [
      {
        path: '/search',
        element: component,
        loader: testLoader,
      },
    ],
    {
      initialEntries: ['/', initialPath],
      initialIndex: 1,
    }
  );
  const wrappedComponent = (
    <Provider store={testStore}>
      <RouterProvider router={testRouter}></RouterProvider>
    </Provider>
  );
  return [wrappedComponent, testRouter];
};
