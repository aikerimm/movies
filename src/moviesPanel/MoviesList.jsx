import MovieCard from './MovieCard';
import React from 'react';
import { useSelector } from 'react-redux';
import { getMoviesData } from '../util/moviesSlice';

const MoviesList = () => {
  const movies = useSelector(getMoviesData);
  return (
    <>
      <p className='moviesCounter'>{movies.totalAmount} movies found</p>
      <div className='moviesList'>
        {movies.data.map((movie) => (
          <MovieCard key={movie.id} movie={movie} />
        ))}
      </div>
    </>
  );
};

export default React.memo(MoviesList);
