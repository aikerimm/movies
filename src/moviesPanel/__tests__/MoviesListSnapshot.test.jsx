/**
 * @jest-environment jsdom
 */

import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import MoviesList from '../MoviesList.jsx';
import moviesReducer from '../../util/moviesSlice.jsx';
import { configureStore } from '@reduxjs/toolkit';
import { testMovie } from '../../util/test-util.jsx';
import { BrowserRouter } from 'react-router-dom';

it('MovieCard renders correctly', () => {
  const testStore = configureStore({
    reducer: { movies: moviesReducer },
    preloadedState: {
      movies: {
        data: { totalAmount: 1, data: [testMovie], offset: 0, limit: 12 },
        totalAmount: 1,
      },
    },
  });

  const tree = renderer
    .create(
      <Provider store={testStore}>
        <BrowserRouter>
          <MoviesList />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
