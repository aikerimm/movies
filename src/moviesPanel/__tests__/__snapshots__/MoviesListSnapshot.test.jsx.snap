// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders correctly 1`] = `
[
  <p
    className="moviesCounter"
  >
    1
     movies found
  </p>,
  <div
    className="moviesList"
    id="moviesList"
  >
    <div
      className="movieCard"
    >
      <img
        alt="moviePoster"
        className="movieCardPoster"
        onClick={[Function]}
        src=""
      />
      <div
        className="titleYear"
      >
        <p
          className="movieCardTitle"
        >
          Zootopia
        </p>
        <p
          className="movieCardYear"
        >
          1994
        </p>
      </div>
      <p
        className="movieCardGenre"
      >
        Drama, Comedy
      </p>
      <img
        alt="contextMenu"
        className="contextMenuIcon"
        id="contextMenuIcon"
        onClick={[Function]}
        src="/contextMenu.svg"
      />
    </div>
  </div>,
]
`;
